libbeat/processors/actions/json_decode_fields.go

```diff
@@ -0,0 +1,91 @@
+package actions
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/processors"
+)
+
+type jsonDecodeFields struct {
+	Fields []string
+}
+
+func init() {
+	processors.RegisterPlugin("json_decode_fields",
+		configChecked(newJSONDecodeFields,
+			requireFields("fields"),
+			allowedFields("fields", "when")))
+}
+
+func newJSONDecodeFields(c common.Config) (processors.Processor, error) {
+	config := struct {
+		Fields []string `config:"fields"`
+	}{}
+	err := c.Unpack(&config)
+	if err != nil {
+		return nil, fmt.Errorf("fail to unpack the json_decode_fields configuration: %s", err)
+	}
+
+	f := jsonDecodeFields{Fields: config.Fields}
+	return f, nil
+}
+
+func (f jsonDecodeFields) Run(event common.MapStr) (common.MapStr, error) {
+	for _, field := range f.Fields {
+		var output map[string]interface{}
+
+		data, err := event.GetValue(field)
+		if err != nil {
+			return event, fmt.Errorf("no %s field found: %s", field, err)
```

Comment:
You must be looking at a stale copy of the drop_fields.go. Here's the [latest](https://github.com/elastic/beats/blob/master/libbeat/processors/actions/drop_fields.go#L50) version that puts the errors into a slice and continues.

> mapstr.GetValue() returns an error when it doesn't find a key. I'm hesitant to use a hasKey + getValue since it will involve parsing the data twice. Any suggestions on how to fix this ?

If you want to ignore errors caused by the key being absent, you can check if the cause is an `common.ErrKeyNotFound` like the [include_fields](https://github.com/elastic/beats/blob/master/libbeat/processors/actions/include_fields.go#L55-L59) action does.